### ISSUE TITLE
Rule Suggestion: Disallow comma operator in switch 'case' clauses

### ISSUE THREAD

**Issue by RyanCavanaugh** [palantir/tslint#1545]
```ts
let arr = [];
switch(arr.length) {
  case 0, 1:
    return 'zero or one';
  default:
    return 'more than one';
}
```

This doesn't behave as expected (see https://github.com/Microsoft/TypeScript/issues/10802) even though it looks like it might

**Comment by 06needhamt** [comment]
This would have saved me hours of pain as shown in https://github.com/Microsoft/TypeScript/issues/10802#issuecomment-245904797

**Comment by 06needhamt** [comment]
This would of saved me from hours of pain as shown in https://github.com/Microsoft/TypeScript/issues/10802#issuecomment-245904797

**Comment by IllusionMH** [comment]
I've started looking on this and I'm curious: should new rule enforce more strict checks for expressions in `case` clauses with set of options?
For example:
* `literals-only` - allows only literals
* `no-comma` - disallows coma operator in expressions
* `no-call-expressions` - option which disables call expressions
* `no-binary-expressions` - option which disables binary expressions

@JKillian, @RyanCavanaugh, @adidahiya, what do you think?

**Comment by JKillian** [comment]
I can't think of a case where I've wanted an complex `switch` pattern. However, `literals-only` might be a bit of overkill, as you couldn't do this:

`
switch(someVar) {
    case SomeEnum.VALUE_1:
        ...
}
```

**Comment by RyanCavanaugh** [comment]
We'll likely take https://github.com/Microsoft/TypeScript/issues/10814 so probably this rule won't be needed

**Comment by JKillian** [comment]
I'm going to close this issue since TS is likely picking this up. If things change on the TS side (or even if it's just going to be a long time before this is implemented) this would be a great rule and we should implement it
